.github/workflows/e2e.yaml

```diff
@@ -143,79 +152,83 @@ jobs:
-    #     # .net.
-    #     # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
-    #     # otherwise pods stops trying to resolve the domain.
-    #     # Get the current config
-    #     original_coredns=$(/usr/local/bin/kubectl get -oyaml -n=kube-system configmap/coredns)
-    #     echo "Original CoreDNS config:"
-    #     echo "${original_coredns}"
-    #     # Patch it
-    #     fixed_coredns=$(
-    #       printf '%s' "${original_coredns}" | sed \
-    #         -e 's/^.*kubernetes cluster\.local/& net/' \
-    #         -e '/^.*upstream$/d' \
-    #         -e '/^.*fallthrough.*$/d' \
-    #         -e '/^.*forward . \/etc\/resolv.conf$/d' \
-    #         -e '/^.*loop$/d' \
-    #     )
-    #     echo "Patched CoreDNS config:"
-    #     echo "${fixed_coredns}"
-    #     printf '%s' "${fixed_coredns}" | /usr/local/bin/kubectl apply -f -
-
-    # - name: Run tests
-    #   run: |
-    #     export KUBERNETES_CONFORMANCE_TEST='y'
-    #     export E2E_REPORT_DIR=${PWD}/_artifacts
-
-    #     # Run tests
-    #     /usr/local/bin/ginkgo --nodes=25                \
-    #       --focus="\[Conformance\]"     \
-    #       --skip="Feature|Federation|machinery|PerformanceDNS|DualStack|Disruptive|Serial|Slow|KubeProxy|LoadBalancer|GCE|Netpol|NetworkPolicy|NodeConformance"   \
-    #       /usr/local/bin/e2e.test                       \
-    #       --                                            \
-    #       --kubeconfig=${PWD}/_artifacts/kubeconfig.conf     \
-    #       --provider=local                              \
-    #       --dump-logs-on-failure=false                  \
-    #       --report-dir=${E2E_REPORT_DIR}                \
-    #       --disable-log-dump=true
-
-    # - name: Upload Junit Reports
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-junit-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: './_artifacts/*.xml'
-
-    # - name: Export logs
-    #   if: always()
-    #   run: |
-    #     /usr/local/bin/kind export logs --name ${KIND_CLUSTER_NAME} --loglevel=debug ./_artifacts/logs
-
-    # - name: Upload logs
-    #   if: always()
-    #   uses: actions/upload-artifact@v2
-    #   with:
-    #     name: kind-logs-${{ env.JOB_NAME }}-${{ github.run_id }}
-    #     path: ./_artifacts/logs
+    - name: Get Cluster status
+      run: |
+        # wait network is ready
+        sleep 5
+        /usr/local/bin/kubectl get nodes -o wide
+        /usr/local/bin/kubectl get pods -A
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods --namespace=kube-system -l k8s-app=kube-dns
+        # smoke test
+        /usr/local/bin/kubectl run test --image httpd:2
+        /usr/local/bin/kubectl wait --timeout=1m --for=condition=ready pods test
+        /usr/local/bin/kubectl get pods -A -owide
+        if [ ${{ matrix.ipFamily }} = "ipv4" ];then 
+          sleep 50s
+        fi
```

Comment:
```shell
...
2024-01-25T03:45:44.534421586Z stderr F I0125 03:45:44.534167       1 options.go:146] SyncForever set to false.
2024-01-25T03:45:44.552060196Z stderr F E0125 03:45:44.551978       1 clientset.go:188] "cannot connect once" err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup konnectivity-server.kube-system.svc.cluster.local on 10.96.0.10:53: no such host\""
2024-01-25T03:45:49.612075002Z stderr F E0125 03:45:49.611973       1 clientset.go:188] "cannot connect once" err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup konnectivity-server.kube-system.svc.cluster.local on 10.96.0.10:53: no such host\""
2024-01-25T03:45:57.263598167Z stderr F E0125 03:45:57.263510       1 clientset.go:188] "cannot connect once" err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup konnectivity-server.kube-system.svc.cluster.local on 10.96.0.10:53: no such host\""
2024-01-25T03:46:09.62534014Z stderr F E0125 03:46:09.625241       1 clientset.go:188] "cannot connect once" err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup konnectivity-server.kube-system.svc.cluster.local on 10.96.0.10:53: no such host\""
2024-01-25T03:46:27.877018898Z stderr F I0125 03:46:27.876930       1 client.go:210] "Connect to server" serverID="d8477c2e-b773-4c1e-be78-8cf8caa66ce2"
2024-01-25T03:46:27.877040759Z stderr F I0125 03:46:27.876951       1 clientset.go:222] "sync added client connecting to proxy server" serverID="d8477c2e-b773-4c1e-be78-8cf8caa66ce2"
2024-01-25T03:46:27.877080614Z stderr F I0125 03:46:27.876986       1 client.go:321] "Start serving" serverID="d8477c2e-b773-4c1e-be78-8cf8caa66ce2" agentID="ba7e8067-df06-45e4-94f4-c75333c559d5"
```

This is a flaky problem. The ANP agent cannot access the service normally within tens of seconds. see the log of ANP agent ^^

I didn't find the root cause, so I simply dealt with it using a temporary method.

I've tested 30s and it didn't work, i tested 15 times with 50s without problems, so it's a safe value.

Reply:
Use readiness of ANP agent and not sleep, Thanks @aojea :)

Reply:
thanks to you for fixing it 🥇